vochain/scrutinizer/scrutinizer.go

```diff
@@ -34,16 +38,19 @@ func (s *Scrutinizer) AddEventListener(l EventListener) {
 type Scrutinizer struct {
 	VochainState *vochain.State
 	// votePool is the list of votes that will be added on the current block
-	votePool []*models.Vote
+	votePool map[string][]*models.Vote
 	// newProcessPool is the list of new process IDs on the current block
 	newProcessPool []*types.ScrutinizerOnProcessData
 	// updateProcessPool is the list of process IDs that require sync with the state database
 	updateProcessPool [][]byte
 	// resultsPool is the list of processes that finish on the current block
 	resultsPool []*types.ScrutinizerOnProcessData
+	// list of live processes (those on which the votes will be computed on arrival)
+	liveResultsProcs sync.Map
 	// eventListeners is the list of external callbacks that will be executed by the scrutinizer
 	eventListeners []EventListener
 	db             *badgerhold.Store
+	addVoteLock    sync.RWMutex
```

Comment:
That's for avoiding the "Transaction Conflict" error which happens quite often. That's a non critical error and the badger developers recommend to try it again until the db transaction works. But I preferred to try to avoid directly the put/get conflict with a Mutex. And yes, with that Mutex the iteration over the "Transaction Conflict" error might be removed but I let it there for security since I'm still not 100% sure all conflicting calls are mutex.

Reply:
If retrying works, then why have the mutex? I don't know, having a mutex around the entire DB feels hacky. If we have good tests and they continuously pass without the mutex and the error retry, then I'd say that's good.

Reply:
Well, to me it looks that repeating a DB transaction N times until it success is kind of hacky. Using a Mutex on this scope makes sense and will release load on the KV context. 

In addition I added a `maxRetries=1000` mechanism which is kind of arbitrary (I tested different values until there were no more transactions failing). This value works in my local computer but a slower one could need more maxRetries. I don't like to block the goroutine forever if there is an issue on the KV (that won't accept new transactions) I prefer instead to try N times and raise an Error. So this is why I'm mixing both approaches: Mutex + Repeat Transaction N times

I'll add a to-do for removing it in the future, with proper research and testing involved. But for now I prefer to make it safe.